PayButton/paybutton-server
# [Issue #248] fix: redis flush and jobs repeating

Description
---
Part of #248.
-  Make so that some fail jobs are retried,
-  Removes unnecessary parameter for creating Payments,
- Fixes flushing jobs cache when starting the redis container

Test plan
---
The dashboard should still work normally.

Besides that, to test that the jobs are being rerun when failed:

1. Save the following content into a filed named `temp.patch` on the repo root:
```
diff --git a/jobs/workers.ts b/jobs/workers.ts
index b08a0b4..5acf1c7 100644
--- a/jobs/workers.ts
+++ b/jobs/workers.ts
@@ -40,6 +40,7 @@ const syncAndSubscribeAddressList = async (addressList: Address[]): Promise<void
 
 const syncAllAddressTransactionsForNetworkJob = async (job: Job): Promise<void> => {
   console.log(`job ${job.id as string}: syncing all addresses for network ${job.data.networkId as string}...`)
+  throw new Error("Test")
   try {
     const addresses = await addressService.fetchAllAddressesForNetworkId(job.data.networkId)
     await syncAndSubscribeAddressList(addresses)

```
2. Run `patch -p1 < temp.patch` (this is to make the jobs give an error)
3. Rerun the jobs with `yarn docker jr`
3. Inspect the `jobs/out.log` file. It should give errors when trying to sync the network 1, then another error when syncing network 2. Three times.
